cmd/abc/abc.go

```diff
@@ -136,31 +138,63 @@ func setLogEnvVars() {
 	}
 }
 
-func realMain(ctx context.Context) error {
-	if err := checkSupportedOS(); err != nil {
-		return err
-	}
-
+func checkVersion(ctx context.Context) func() {
 	// Only check for updates if not built from HEAD.
 	if version.Version != "source" {
```

Comment:
I know this was already like this, but we can invert this conditional and save one level of nesting.

```
if version.Version == "source" {
    return func(){}
```

Reply:
Good call